### PR TITLE
[action] Add force-with-lease to push_to_git_remote

### PR DESCRIPTION
**Pull request by sprabhuanand**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

I need to have the git option --force-with-lease, so that my changes to git will be pushed only when both local and remote is same. Force option sometimes overwrites the code when the remote is ahead of local

I ran the bundle exec fastlane test command and made sure all actions passed. I have pasted the output below

### Description

+------+---------------------------------------------------------------+-------------+
|                                  fastlane summary                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1    | Switch to validate_repo lane                                  | 0           |
| 2    | Switch to lint_source lane                                    | 0           |
| 3    | ensure_no_debug_code                                          | 0           |
| 4    | ensure_no_debug_code                                          | 0           |
| 5    | ensure_no_debug_code                                          | 0           |
| 6    | ensure_no_debug_code                                          | 0           |
| 7    | ensure_no_debug_code                                          | 0           |
| 8    | ensure_no_debug_code                                          | 0           |
| 9    | rubocop                                                       | 1           |
| 10   | find -E . -regex ".*\.(sh|bash)" -not -name "Pods-*" -name ". | 0           |
| 11   | Switch to execute_tests lane                                  | 0           |
| 12   | Switch to local_version lane                                  | 0           |
| 13   | PAGER=cat bin/fastlane --help                                 | 0           |
| 14   | bundle check || bundle install                                | 0           |
| 15   | rake install                                                  | 15          |
| 16   | bundle exec rake test_all                                     | 201         |
| 17   | Switch to validate_docs lane                                  | 0           |
| 18   | git clone https://github.com/fastlane/docs --depth=1          | 13          |
| 19   | Switch to verify_docs lane                                    | 0           |
| 20   | git clone https://github.com/fastlane/docs --depth=1          | 20          |
| 21   | bundle install                                                | 0           |
| 22   | bundle exec fastlane test skip_building:true                  | 3           |
| 23   | Switch to ensure_tool_name_formatting lane                    | 0           |
| 24   | Switch to ensure_code_samples lane                            | 0           |
| 25   | test_sample_code                                              | 0           |
| 26   | Switch to ensure_code_snippets lane                           | 0           |
| 27   | Switch to ensure_actions_config_items_formatting lane         | 0           |
+------+---------------------------------------------------------------+-------------+

### CODE REVIEW

**Comment by sprabhuanand** [comment]
I signed it!

File: fastlane/lib/fastlane/actions/push_to_git_remote.rb [hunk 1]
```diff
@@ -52,6 +55,11 @@ def self.available_options
                                        description: "Force push to remote",
                                        is_string: false,
                                        default_value: false),
+          FastlaneCore::ConfigItem.new(key: :force_with_lease,
+                                       env_name: "FL_PUSH_GIT_FORCE_WITH_LEASE",
+                                       description: "Force push with lease to remote",
+                                       is_string: false,
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/push_to_git_remote.rb]: Can we change this line to `type: Boolean` instead? 😊

Comment (joshdholtz) [reply fastlane/lib/fastlane/actions/push_to_git_remote.rb]: And while you are in there, can you change the other `is_string: false` in the other two options to `type: Boolean` as well? 😊

Comment (joshdholtz) [reply fastlane/lib/fastlane/actions/push_to_git_remote.rb]: `type: Boolean` is preferred so the `Fastlane.swift` knows what type of arguments are needed

**Review by joshdholtz — CHANGES_REQUESTED** [review]

**Comment by joshdholtz** [comment]
@sprabhuanand Just wanted to ping you to see if you could make these changes? 😊

**Comment by prabhuanand** [comment]
I signed it!

**Review by joshdholtz — APPROVED** [review]
Looks great! Thanks for making those changes 😊

**Comment by janpio** [comment]
`master` is broken after this merged:

```
[10:27:05]: Found errors in the documentation, more information below
[10:27:05]: Error found in code sample:
push_to_git_remote # simple version. pushes "master" branch to "origin" remote
push_to_git_remote(
remote: "origin",         # optional, default: "origin"
local_branch: "develop",  # optional, aliased by "branch", default: "master"
remote_branch: "develop", # optional, default is set to local_branch
force: true,              # optional, default: false
force_with_lease: true,   # optional, default: false
tags: false               # optional, default: true
)
'force' value must be a Fastlane::Boolean! Found TrueClass instead.
```
https://circleci.com/gh/fastlane/fastlane/24028?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Pull request merged**